ddtrace/_trace/tracer.py

```diff
@@ -224,11 +224,14 @@ def __init__(self) -> None:
             log.debug("Failed to store the configuration on disk", extra=dict(error=e))
 
     @property
-    def _agent_url(self):
-        return getattr(self._span_aggregator.writer, "intake_url", None)
+    def _agent_url(self) -> Optional[str]:
+        if isinstance(self._span_aggregator.writer, (HTTPWriter, AgentWriterInterface)):
+            # For AgentWriterInterface, we can return the intake_url directly
+            return self._span_aggregator.writer.intake_url
+        return None
 
     @_agent_url.setter
-    def _agent_url(self, value):
+    def _agent_url(self, value: str):
         if isinstance(self._span_aggregator.writer, HTTPWriter):
```

Comment:
Do we have to update this `isinstance` too to include `AgentWriterInterface`?

Reply:
I don't think we *have* to, but good call out / good idea.

```suggestion
        if isinstance(self._span_aggregator.writer, (HTTPWriter, AgentWriterInterface)):
```